### PR TITLE
Additional security text about on-link provisioning protocol security considerations

### LINKED ISSUE #30: security review: DHCP/RA injecting attacker discussion
Thread:

  * https://mailarchive.ietf.org/arch/msg/captive-portals/-HpoBCzx8LlrzU4z9mcrVShWdQs/

### PR DESCRIPTION
**Pull request by ekline**
Fixes: #30

### CODE REVIEW

File: draft-ietf-capport-rfc7710bis.xml [hunk 1]
```diff
@@ -356,14 +356,27 @@
...
       reassure the user that the captive portal is not malicious.</t>
 
+      <t>Each of the options described in this document is presented to a node
+      using the same protocols used to provision other information critical
+      to the node's successful configuration on a network.  The security
+      considerations applicable to each of these provisioning mechanisms also
+      apply when the node is attempting to learn the information conveyed in
+      these options. In the absence of security measures like RA Guard
+      (<xref target="RFC6105"/>, <xref target="RFC7113"/>) or DHCP Shield
+      <xref target="RFC7610"/>, an attacker may have the ability to inject,
```
Comment (martinthomson) [inline draft-ietf-capport-rfc7710bis.xml]:
```suggestion
      <xref target="RFC7610"/>, an attacker could inject,
```

File: draft-ietf-capport-rfc7710bis.xml [hunk 2]
```diff
@@ -356,14 +356,27 @@
...
       <t>An attacker with the ability to inject DHCP messages or RAs
       could include an option from this document to force users to contact
       an address of his choosing. As an attacker with this capability could
       simply list himself as the default gateway (and so intercept all the
       victim's traffic); this does not provide them with significantly more
       capabilities, but because this document removes the need for
       interception, the attacker may have an easier time performing the
-      attack. As the operating systems and application that make use of this
+      attack. The use of an HTTPS URI with an IP address string literal in this
+      environment would of course add no significant security.</t>
```
Comment (martinthomson) [inline draft-ietf-capport-rfc7710bis.xml]: I don't see how this is an improvement.  The difference between a name and an address is only in the potential for the DNS provider to get involved.  If you believe in the using DNS to watch for Bad Stuff (this is a contentious topic), then a name has the advantage that you have another gatekeeper involved.  Of course, the attacker likely just set the DNS server, so they can equally avoid that gatekeeper too.

But that's not what is going on here.  You seem to be implying that there might be some understanding that an IP literal is somehow different with respect to a name, only to suggest that the difference is obvious.

Unless I'm missing something, I'd suggest that you strike this new sentence.  If your point is that there is no difference between an IP literal and a name (which is generally - but not completely - true by my reckoning), that risks getting into the debate about use of DNS resolvers other than the one offered by the network.

**Review by martinthomson — COMMENTED** [review]
The new paragraph is good.  The other sentence is a little hard to understand and risks opening questions that we probably don't want opened.

File: draft-ietf-capport-rfc7710bis.xml (hunk 2 above)
Comment (ekline) [reply draft-ietf-capport-rfc7710bis.xml]: I had been attempting to incorporate the feedback from Rifaat.

I guess allocation of certs for IP addresses generally follows the same rigour as it does for FQDNs.

Comment (ekline) [reply draft-ietf-capport-rfc7710bis.xml]: Removed.

File: draft-ietf-capport-rfc7710bis.xml (hunk 1 above)
Comment (ekline) [reply draft-ietf-capport-rfc7710bis.xml]: Fixed.

**Pull request merged**